### ISSUE TITLE
Serializer & KeySerializer should be Serializable

### ISSUE THREAD

**Issue by ssimeonov** [json4s/json4s#680]
Since `Formats` extends `Serializable`, for consistency, custom serializers should be serializable as well. The suggestion is to make `Serializer` and `KeySerializer` extend `Serializable` as well.

For context, the most active Scala OSS project--Apache Spark--works on the basis of serializing functions and their closures. As an added twist, > 100K data engineers and data scientists work with Spark in notebook environments, which are essentially Scala REPL frontends. The REPL uses closures aggressively and that doesn't mix well with the json4s implicit `Formats` design. While using `@transient implicit val` can address this problem when JSON processing is not part of the distributed Spark behavior, it fails when `Formats` is needed on the workers.

In other words, this issue is not just about a consistency fix, but about a problem that affects a very large  base of users who are not particularly skilled in Scala & JVM internals. The risk of the fix seems low given the benefit.

**Comment by magnolia-k** [comment]
Can you give us a specific example of the code?

**Comment by ssimeonov** [comment]
@magnolia-k it is as simple as

```scala
  trait Serializer[A] extends Serializable {
    // ...
  }

  trait KeySerializer[A] extends Serializable  {
    // ...
  }
```
